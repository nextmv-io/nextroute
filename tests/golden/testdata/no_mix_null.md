# Missing values in no mix constraint (no_mix_null.json)

This example demonstrates missing values when defining a _no mix_ constraint.
I.e., some of the stops either have no `mixing_items` defined or have it set to
`null`. These stops can be planned anywhere in a route, independent of the
current mix of items in the vehicle.

Find some notes about the example below:

- There is only one vehicle servicing all the stops.
- Stops `north` and `south` belong to mixing group `A`.
- Stops `east` and `west` belong to mixing group `B`.
- The transports from north to south and from east to west cannot overlap. This
  means that a route like `north -> east -> south -> west` is not feasible.
- All other stops can go anywhere in the route and should be planned in a way
  that minimizes the total travel time.
- All stops are located on a circle with their names indicating the cardinal
  direction they are located at. This is useful for checking the time-efficiency
  of the solution.
