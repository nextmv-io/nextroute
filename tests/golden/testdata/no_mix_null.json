{
  "stops": [
    {
      "id": "north",
      "location": {
        "lon": 7.6256,
        "lat": 51.9714
      },
      "precedes": "south",
      "mixing_items": {
        "main": {
          "name": "A",
          "quantity": 1
        }
      }
    },
    {
      "id": "south",
      "location": {
        "lon": 7.6256,
        "lat": 51.9534
      },
      "precedes": "south_west",
      "mixing_items": {
        "main": {
          "name": "A",
          "quantity": -1
        }
      }
    },
    {
      "id": "east",
      "location": {
        "lon": 7.6402,
        "lat": 51.9624
      },
      "precedes": "west",
      "mixing_items": {
        "main": {
          "name": "B",
          "quantity": 1
        }
      }
    },
    {
      "id": "west",
      "location": {
        "lon": 7.611,
        "lat": 51.9624
      },
      "precedes": "north_west",
      "mixing_items": {
        "main": {
          "name": "B",
          "quantity": -1
        }
      }
    },
    {
      "id": "north_east",
      "location": {
        "lon": 7.6359,
        "lat": 51.9688
      },
      "mixing_items": null
    },
    {
      "id": "south_east",
      "location": {
        "lon": 7.6359,
        "lat": 51.956
      },
      "mixing_items": null
    },
    {
      "id": "south_west",
      "location": {
        "lon": 7.6153,
        "lat": 51.956
      }
    },
    {
      "id": "north_west",
      "location": {
        "lon": 7.6153,
        "lat": 51.9688
      }
    }
  ],
  "vehicles": [
    {
      "id": "truck",
      "speed": 20
    }
  ]
}
